core/domain/exp_services.py

```diff
@@ -535,12 +536,8 @@ def _save_exploration(committer_id, exploration, commit_message, change_list):
         Exception: The versions of the given exploration and the currently
             stored exploration model do not match.
     """
-    exploration_rights = rights_manager.get_exploration_rights(exploration.id)
-    if exploration_rights.status != rights_manager.ACTIVITY_STATUS_PRIVATE:
-        exploration.validate(strict=True)
-    else:
-        exploration.validate()
-
+    exploration.validate(
+        strict=rights_manager.is_exploration_public(exploration.id))
```

Comment:
I really suggest not doing this -- things start to become a bit hard to parse when you combine so many different concepts into a single line of code. Perhaps take the `rights_manager.is_exploration_public(exploration.id)` part and define a local `exploration_is_public` variable.

There is value in the intermediate variable: its name gives context about what is being calculated. This may not be so apparent here but is more apparent in places below like `state_names_without_classifier`.